src/main/java/org/terasology/cities/common/LineSegment2.java

```diff
@@ -0,0 +1,283 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.cities.common;
+
+import org.joml.Vector2f;
+import org.joml.Vector2fc;
+
+//TODO: move to joml-ext
```

Comment:
:+1: I think it's okay to keep this in here for now to get it migrated, and then align `LineSegment2/3f/d` in a final step.